nilearn/_utils/testing.py

```diff
@@ -161,23 +161,25 @@ def write_tmp_imgs(*imgs, **kwargs):
             with warnings.catch_warnings():
                 warnings.simplefilter("ignore", RuntimeWarning)
                 for img in imgs:
-                    filename = tempfile.mktemp(prefix=prefix,
-                                               suffix=suffix,
-                                               dir=None)
-                    filenames.append(filename)
+                    fd, filename = tempfile.mkstemp(prefix=prefix,
```

Comment:
sorry if my previous comment was unclear, but you need to close the file right away and return the filename only
otherwise you yield the filename but retain an open file handle and windows won't allow opening the file for writing or removing it.
(the file is not removed when you close it)
you don't need to store the file descriptor once you have closed the file